uui/components/filters/FilterDatePickerBody.tsx

```diff
@@ -1,58 +1,98 @@
-import React from 'react';
-import dayjs from 'dayjs';
+import React, {
+    Fragment, useEffect, useState,
+} from 'react';
+import dayjs, { Dayjs } from 'dayjs';
 import { i18n } from '../../i18n';
-import { DatePickerCoreProps, IDropdownBodyProps } from '@epam/uui-core';
-import { BaseDatePicker } from '@epam/uui-components';
-import { FlexSpacer, FlexRow, FlexCell } from '../layout';
+import { IDropdownBodyProps, useUuiContext } from '@epam/uui-core';
+import {
+    FlexSpacer, FlexRow, FlexCell,
+} from '../layout';
 import { LinkButton } from '../buttons';
 import { Text } from '../typography';
-import { DatePickerBody } from '../datePickers';
+import {
+    DatePickerBody, DatePickerProps, DatePickerBodyValue, ViewType,
+} from '../datePickers';
+import { getNewMonth } from '../datePickers/helpers';
 
-/** Represents the properties of the DatePicker. */
-export interface DatePickerProps extends DatePickerCoreProps, IDropdownBodyProps {}
+/**
+ * Represents the properties of the FiterDatePicker
+ */
+export interface FilterDatePickerProps extends DatePickerProps, IDropdownBodyProps {}
 
-export class FilterDatePickerBody extends BaseDatePicker<DatePickerProps> {
-    renderInput = (): any => {
-        return null;
-    };
+export function FilterDatePickerBody(props: FilterDatePickerProps) {
+    const { value } = props;
+    const context = useUuiContext();
+
+    const [{
+        view,
+        month,
+    }, setState] = useState<{
+        month: Dayjs;
+        view: ViewType;
+    }>({
+        view: 'DAY_SELECTION',
+        month: getNewMonth(value),
+    });
+
+    useEffect(() => {
+        setState((prev) => ({
+            ...prev,
+            month: getNewMonth(value),
+        }));
+    }, [value]);
 
-    onToggleHandler = (val: boolean) => {
-        this.onToggle(val);
-        this.props.onClose();
+    const handleValueChange = (newValue: string | null) => {
+        props.onValueChange(newValue);
+
+        if (props.getValueChangeAnalyticsEvent) {
+            const event = props.getValueChangeAnalyticsEvent(newValue, value);
+            context.uuiAnalytics.sendEvent(event);
+        }
+        if (newValue) {
+            props.onClose?.();
```

Comment:
Won't we lose close of picker after clear, when value will be nullish?

Reply:
We can't do that actually in FilterDatePicker, but anyway will remove this condition